python/pyspark/context.py

```diff
@@ -324,21 +359,23 @@ def _do_init(
             self.profiler_collector = None
 
         # create a signal handler which would be invoked on receiving SIGINT
-        def signal_handler(signal, frame):
+        def signal_handler(signal: Any, frame: Any) -> None:
```

Comment:
Might be overkill, but you can take a look at

https://github.com/python/typeshed/blob/ee487304d76c671aa353a15e34bc1102bffa2362/stdlib/signal.pyi#L188

and

https://github.com/python/typeshed/blob/ee487304d76c671aa353a15e34bc1102bffa2362/stdlib/signal.pyi#L83

Also it might be `NoReturn` in our case.